### ISSUE TITLE
ExtStorage broken in 2.10

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#835]
Originally reported of Google Code with ID 779.

```
ExtStorage is currently broken in 2.10, with the arbitrary arguments not passing validation:

$ gnt-instance add --disk-template ext --disk=0:size=15G,provider=3par,param1="full" --os-type debootstrap+default --hypervisor-parameters kvm:initrd_path=,kernel_path=
--net 0:link=cmstest lintest42

2014-03-26 14:28:58,937: ganeti-masterd pid=4354/Jq8/Job41095 INFO Op 1/1: opcode INSTANCE_CREATE(lintest42) waiting for locks
2014-03-26 14:28:58,939: ganeti-masterd pid=4354/Jq8/Job41095/I_CREATE ERROR OpCode OP_INSTANCE_CREATE, parameter disks, has invalid type <type 'list'>/value '[{'size':
15360, 'param1': 'full', 'provider': '3par'}]' expecting type List of (Dictionary with keys of (OneOf name, vg, adopt, spindles, mode, provider, metavg, size) and values of
(NonEmptyString or Integer) [Disk parameters])
2014-03-26 14:28:58,939: ganeti-masterd pid=4354/ClientReq2 INFO Received job poll request for 41095
2014-03-26 14:28:58,939: ganeti-masterd pid=4354/Jq8/Job41095 ERROR Op 1/1: Caught exception in INSTANCE_CREATE(lintest42)
Traceback (most recent call last):
  File "/usr/share/ganeti/2.10/ganeti/jqueue.py", line 1121, in _ExecOpCodeUnlocked
    timeout=timeout)
  File "/usr/share/ganeti/2.10/ganeti/jqueue.py", line 1432, in _WrapExecOpCode
    return execop_fn(op, *args, **kwargs)
  File "/usr/share/ganeti/2.10/ganeti/mcpu.py", line 528, in ExecOpCode
    lu = lu_class(self, op, self.context, self.rpc)
  File "/usr/share/ganeti/2.10/ganeti/cmdlib/base.py", line 122, in __init__
    self.op.Validate(True)
  File "/usr/share/ganeti/2.10/ganeti/opcodes_base.py", line 247, in Validate
    errors.ECODE_INVAL)
OpPrereqError: ("Parameter 'OP_INSTANCE_CREATE.disks' fails validation", 'wrong_input')
2014-03-26 14:28:58,949: ganeti-masterd pid=4354/Jq8/Job41095 INFO Finished job 41095, status = error

See Debian bug #742699: http://bugs.debian.org/742699

It looks a lot like what 2e09344f461245c0b7f25cf7da51ab386a0807ae fixed for 2.11.
```
Originally added on 2014-03-26 16:16:01 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Sounds awfully familiar. Can you please check if simply cherry-picking b26a275a0
solves the problem?

As in 2.10 jobs are still handled by masterd, the mentioned commit 2e09344
might not be necessary to be cherry-picked, as the parameter stays within
the Python part.

But if this is not sufficient, I'm happy with cherry-picking b26a275a0 and 2e09344f with its
prerequisites.
```
Originally added on 2014-03-26 16:43:35 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Cherry-picking b26a275a0 indeed solves the problem.

Thanks,
Apollon
```
Originally added on 2014-03-27 14:15:51 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit f22348c63ddfbef6704e3abd07ef94862f0df33c
Author: Klaus Aehlig <aehlig@google.com>
Date:   Mon Feb 3 17:50:11 2014 +0100

    Fix specification of TIDiskParams
    
    Commit 580b1fdd incorrectly assumes that disk parameters are
    just the standard ones, whereas the man page explicitly states
    that additional parameters can be passed as well, if they make
    sense for the chosen storage type. Fix this.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>
    
    Cherry-picked from b26a275a0. Fixes issue 779.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Hrvoje Ribicic <riba@google.com>

```
Originally added on 2014-03-27 14:35:55 +0000 UTC.

Mentioned Issue with GoogleCode ID 779 has been migrated to GitHub with ID #835.
Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released in 2.10.3.
```
Originally added on 2014-04-16 10:37:20 +0000 UTC.

Changed State: Released
